src/components/Pages/LoginPage/LoginPage.tsx

```diff
@@ -126,22 +127,13 @@ export const LoginPage = () => {
               // If the connected account does not have a profile, redirect the user to the setup page to create a new one.
               // The setup page should then redirect the user to the url provided as query param if available.
               if (!profile) {
-                window.location.href = '/auth/setup' + (redirectTo ? `?redirectTo=${redirectTo}` : '')
-                setShowConnectionModal(false)
-                return
+                navigate(locations.setup(redirectTo))
+                return setShowConnectionModal(false)
```

Comment:
I know this was made like this before, but maybe it can be removed, the component will be unmounted on navigation and the state reset